src/compose/application-manager.ts

```diff
@@ -788,12 +786,19 @@ function reportOptionalContainers(serviceNames: string[]) {
 		'. ',
 	)}`;
 	log.info(message);
-	return logger.logSystemMessage(
-		message,
-		{},
-		'optionalContainerViolation',
-		true,
+	return logger.logSystemMessage(message, {});
+}
+
+function reportRejectedReleases(violators: contracts.ContractViolators) {
+	const appStrings = Object.values(violators).map(
+		({ appName, services }) =>
+			`${appName}: Services with unmet requirements: ${services.join(', ')}`,
 	);
+	const message = `Some releases were rejected due to having unmet requirements:\n  ${appStrings.join(
+		'\n  ',
+	)}`;
+	log.error(message);
+	return logger.logSystemMessage(message, { error: true });
```

Comment:
This seems like it'd be a `void` return that might be a leftover from coffeescript? I'm expecting the `return` can just be removed:
```suggestion
	logger.logSystemMessage(message, { error: true });
```
?